src/submission/submission-to-clinical/merge-submission.ts

```diff
@@ -54,10 +62,85 @@ export const mergeActiveSubmissionWithDonors = async (
   return updatedDonors;
 };
 
-const updateSpecimenRecord = (donor: Donor, record: ClinicalEnitityRecord) => {
-  // Find specimen in donor
+// This function will return a merged donor of records mapped by clinical type and the DB exsistentDonor
+export const mergeRecordsMapIntoDonor = (
```

Comment:
I decided to leave it like this since for record -> donor merge I need the treatment to happen before therapies, we can refactor this to be more generic if needed later